doc/source/timeseries.rst

```diff
@@ -889,9 +889,31 @@ The key features of a ``DateOffset`` object are:
 * It can be multiplied by an integer (positive or negative) so that the
   increment will be applied multiple times.
 * It has :meth:`~pandas.DateOffset.rollforward` and
-  :meth:`~pandas.DateOffset.rollback` methods for moving a date forward or 
+  :meth:`~pandas.DateOffset.rollback` methods for moving a date forward or
   backward to the next or previous "offset date".
 
+Beware that adding an offset to a ``DatetimeIndexes`` with consecutive values
+can produce a ``DatetimeIndexes`` with gaps, for example:
+
+.. ipython:: python
+
+index = pd.DatetimeIndex(start='2018-02-27', periods=3, freq='D')
+index
+
+.. ipython:: python
+
+new_index = index + pd.tseries.offsets.DateOffset(months=4, days=2)
+new_index
+
+A way to achieve ``DatetimeIndexes`` without gaps is to shift the first date
+with ``DateOffset`` then construct the ``DatetimeIndexes`` using the original
+frequency:
+
+.. ipython:: python
+start_date = index[0] + pd.tseries.offsets.DateOffset(months=4, days=2)
+new_index = pd.DatetimeIndex(start=start_date, periods=3, freq='D')
```

Comment:
Could you use `date_range` instead of `DatetimeIndex` to construct these indexes? `DatetimeIndex` is valid constructor for the range but `date_range` is a bit more conventional.

There's also another occurrence of this above.